girder/molecules/molecules/molecule.py

```diff
@@ -287,23 +287,22 @@ def conversions(self, output_format, params):
 
         body = self.getBodyJson()
 
-        if 'fileId' not in body:
+        if 'fileId' not in body and 'cjson' not in body:
             raise RestException('Invalid request body.', code=400)
 
-        file_id = body['fileId']
-
-        file = ModelImporter.model('file').load(file_id, user=user)
-
-        input_format = file['name'].split('.')[-1]
-
-        if input_format not in Molecule.input_formats:
-            raise RestException('Input format not supported.', code=400)
+        if 'fileId' in body:
+            file_id = body['fileId']
+            file = ModelImporter.model('file').load(file_id, user=user)
+            input_format = file['name'].split('.')[-1]
 
-        if file is None:
-            raise RestException('File not found.', code=404)
+            if input_format not in Molecule.input_formats:
+                raise RestException('Input format not supported.', code=400)
 
-        with File().load(file) as f:
-            data_str = f.read().decode()
+            with File().open(file) as f:
+                data_str = f.read().decode()
+        else:
+            input_format = 'cjson'
+            data_str = json.dumps(body['cjson'])
```

Comment:
Let check that `cjson` exists and return a 400 status code if we don't have `fileId` or `cjson`.

Reply:
Scratch that, I see the check above ...